register.ts

```diff
@@ -1,30 +1,54 @@
+// tslint:disable callable-types
+// tslint:disable variable-name
 
+declare type NoopDecorator = (
+  _target: object,
+  _propertyKey: string,
+  _descriptor?: PropertyDescriptor
+) => void;
+
+declare var global: NodeJS.Global;
 declare global {
+  /**
+   * Indicates that this function or variable is being overridden
+   * from the implemented or extended parent class.
+   *
+   * @see [TSLint Override](https://github.com/hmil/tslint-override)
+   */
+  var override: NoopDecorator;
 
-    /**
-     * Specifies that this member must override a parent member.
-     */
-    const override: (_target: any, _propertyKey: string, _descriptor?: PropertyDescriptor) => void;
+  /**
+   * Indicates that this function or variable is being overridden
+   * from the implemented or extended parent class.
+   *
+   * @see [TSLint Override](https://github.com/hmil/tslint-override)
+   */
+  var Override: NoopDecorator;
 
-    interface Window {
-        override: (_target: any, _propertyKey: string, _descriptor?: PropertyDescriptor) => void;
-    }
+  interface Window {
+    override: NoopDecorator;
+    Override: NoopDecorator;
+  }
 
-    namespace NodeJS {
-        interface Global {
-            override: (_target: any, _propertyKey: string, _descriptor?: PropertyDescriptor) => void;
-        }
+  interface WorkerGlobalScope {
+    override: NoopDecorator;
+    Override: NoopDecorator;
+  }
+
+  namespace NodeJS {
+    interface Global {
+      override: NoopDecorator;
+      Override: NoopDecorator;
     }
+  }
 }
 
-export const ctx = typeof window !== 'undefined' ?
-    window : global;
+export const ctx =
+  typeof global === 'undefined' ? (typeof self === 'undefined' ? undefined : self) : global;
+export const override: NoopDecorator = () => { /* noop */ };
+export const Override: NoopDecorator = () => { /* noop */ };
 
-/**
- * Specifies that this member must override a parent member.
- */
-export function override(_target: any, _propertyKey: string, _descriptor?: PropertyDescriptor) {
-    // noop
+if (ctx) {
```

Comment:
This is the core of this change, right? By checking that ctx is defined you avoid triggering an error in web workers, but at the cost of making `override` unavailable in this environment. Is that correct?

Also, if you need a solution to this, you might want to look into TypeScript transformers. They are like plugins that you can attach to the compiler to do some modifications of the AST. For instance, you can remove decorators during compilation (angular does that).

As a third alternative to JSDoc and global override symbol, a transformer could be offered to use decorators without registering a global symbol.